*** /usr/bin/mysqld_safe.orig   Fri Nov 21 01:26:39 2014
--- /usr/bin/mysqld_safe        Thu Dec 18 01:35:36 2014
***************
*** 82,87 ****
--- 82,88 ----
    --syslog                   Log messages to syslog with 'logger'
    --skip-syslog              Log messages to error log (default)
    --syslog-tag=TAG           Pass -t "mysqld-TAG" to 'logger'
+   --server-id=auto           Make server_id variable from IP Address 3rd and 4th byte.

  All other options are passed to the mysqld program.

***************
*** 210,215 ****
--- 211,217 ----
        --ledir=*) ledir="$val" ;;
        --malloc-lib=*) set_malloc_lib "$val" ;;
        --mysqld=*) MYSQLD="$val" ;;
+       --server-id=*) SERVER_ID="$val" ;;
        --mysqld-version=*)
          if test -n "$val"
          then
***************
*** 625,630 ****
--- 627,637 ----
    MYSQLD=mysqld
  fi

+ ### Added by yoku0825.
+ if test "$SERVER_ID" = "auto" ; then
+   SERVER_ID=$(getent hosts $(hostname) | awk '{print $1}' | awk -F. '{id=$3 * 256 + $4; print id}')
+ fi
+
  if test ! -x "$ledir/$MYSQLD"
  then
    log_error "The file $ledir/$MYSQLD
***************
*** 760,766 ****
  cmd="`mysqld_ld_preload_text`$NOHUP_NICENESS"

  for i in  "$ledir/$MYSQLD" "$defaults" "--basedir=$MY_BASEDIR_VERSION" \
!   "--datadir=$DATADIR" "--plugin-dir=$plugin_dir" "$USER_OPTION"
  do
    cmd="$cmd "`shell_quote_string "$i"`
  done
--- 767,773 ----
  cmd="`mysqld_ld_preload_text`$NOHUP_NICENESS"

  for i in  "$ledir/$MYSQLD" "$defaults" "--basedir=$MY_BASEDIR_VERSION" \
!   "--datadir=$DATADIR" "--plugin-dir=$plugin_dir" "--server-id=$SERVER_ID" "--report-host=$(uname -n)" "$USER_OPTION"
  do
    cmd="$cmd "`shell_quote_string "$i"`
  done
